spec/requests/users/sessions_spec.rb

```diff
@@ -28,6 +46,32 @@
       expect(assigns(:login_token)).to eq(user.login_token)
       expect(response).to render_template(:redirect_from_magic_link)
     end
+
+    it "redirects to sign in when the token doesn't match" do
+      get "/users/confirm_sign_in?login_token=aaaaaaaaaa"
+
+      expect(response).to redirect_to "/users/sign_in"
+    end
+
+    context "when the token has expired" do
+      before { user.update!(login_token_valid_until: 1.hour.ago) }
+
+      it "redirects to sign in when the token has expired" do
+        get "/users/confirm_sign_in?login_token=#{user.login_token}"
+
+        expect(response).to redirect_to "/users/sign_in"
+      end
+    end
```

Comment:
I think that a context for a single test is a bit excessive.

Especially since we mention that token expired both in context and in test itself.

Reply:
Also, do we want to test for showing an error message?

Reply:
I was just following @amazimbe 's lead with the other tests in this file

Reply:
I think it makes it a lot easier to read actually

Reply:
I'm happy to keep the context, but I don't think we need to duplicate the message "when the token has expired".

Reply:
Ah yes, my mistake, added the context after. Good spot!